packages/@ember/-internals/glimmer/lib/utils/outlet.ts

```diff
@@ -37,27 +55,42 @@ export interface RenderState {
   /**
    * template (the layout of the outlet component)
    */
-  template: Template | TemplateFactory | undefined;
-}
-
-export interface Outlets {
-  [name: string]: OutletState | undefined;
+  template: Template | undefined;
 }
 
 export interface OutletState {
   /**
-   * Nested outlet connections.
+   * Represents what was rendered into this outlet.
    */
-  outlets: Outlets;
+  render: RenderState | undefined;
 
   /**
-   * Represents what was rendered into this outlet.
+   * Represents what, if any, should be rendered into the next {{outlet}} found
+   * at this level.
+   *
+   * This used to be a dictionary of children outlets, including the {{outlet}}
+   * "main" outlet any {{outlet "named"}} named outlets. Since named outlets
+   * are not a thing anymore, this can now just be a single`child`.
    */
-  render: RenderState | undefined;
+  outlets: {
+    main: OutletState | undefined;
+  };
 
   /**
-   * Has to do with render helper and orphan outlets.
-   * Whether outlet state was rendered.
+   * @deprecated
+   *
```

Comment:
I really appreciate all these comments -- as someone who is just learning how the glue fits together, this is _immensely_ helpful for provided much needed context as well as a bit of _why_